File: memory_test/nsquare/nsquare.java
import java.io.File;
import java.io.FileNotFoundException;
import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.Random;
import java.util.Scanner;

public class nsquare {

    static void bubbleSort(int arr[], int n) {
        int i, j, temp;
        boolean swapped;
        for (i = 0; i < n - 1; i++) {
            swapped = false;
            for (j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {

                    temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }

            if (swapped == false)
                break;
        }
    }

    static void printArray(int arr[], int size) {
        int i;
        for (i = 0; i < size; i++)
            System.out.print(arr[i] + " ");
        System.out.println();
    }

    public static void main(String args[]) throws FileNotFoundException {
        Scanner reader = new Scanner(new File("/home/teaguy21/Documents/memory_test/constant/constant.txt"));
        int myNumber = reader.nextInt();
        reader.close();
        int amount = myNumber;
        int[] arr = new int[amount];
        Random rand = new Random();
        for (int i = 0; i < amount; i++) {
            arr[i] = rand.nextInt(amount);
        }
        int n = arr.length;

        long start = System.nanoTime();
        bubbleSort(arr, n);
        long end = System.nanoTime();
        BigDecimal d = new BigDecimal((end - start)/ 1000000000.0).setScale(6, RoundingMode.HALF_EVEN);
        System.out.println(d);
        // System.out.println("Sorted array: ");
        // printArray(arr, n);
    }
}

File: exp/nsquare/nsquare.py
import random
import time


def bubbleSort(arr):
    n = len(arr)

    for i in range(n):
        swapped = False

        for j in range(0, n - i - 1):

            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if swapped == False:
            break

with open("/home/teaguy21/Documents/exp/constant.txt", "r") as file:
  first_line = file.readline()
  number_string = first_line.rstrip("\n")
  number = int(number_string)

input_array = []
amount = number
for i in range(amount):
    input_array.append(random.randint(1, amount))

start = time.time_ns()
bubbleSort(input_array)
end = time.time_ns()

print(format((end - start) / 10 ** 9, ".6f"))
